src/main/java/rick/domain/Point.java

```diff
@@ -4,17 +4,38 @@
 
 public class Point {
 
-    public static final Point MOVABLE_POINT = new Point(true);
-    public static final Point NON_MOVABLE_POINT = new Point(false);
+    private static final String MESSAGE_UNI_DIRECTION_ONLY = "양 방향으로 갈 수 없습니다.";
 
-    private final boolean value;
+    private final boolean left;
+    private final boolean right;
 
-    public Point(boolean value) {
-        this.value = value;
+    public Point(boolean left, boolean right) {
+        if (left && right) {
+            throw new IllegalArgumentException(MESSAGE_UNI_DIRECTION_ONLY);
+        }
+
+        this.left = left;
+        this.right = right;
+    }
+
+    public Point next(boolean nextRight) {
+        if (this.right) {
+            return new Point(this.right, false);
+        }
+
+        return new Point(this.right, nextRight);
     }
```

Comment:
this.right 값을 그대로 사용하는것보단 아래와 같이 방향에 대한 값을 명시적으로 나타내주는것이 가독성이 더 좋다고 생각해요 😄 
더 나아가서는 정적 팩토리 메소드로 정의하는것도 좋아보입니다 😄 

```suggestion
    public Point next(boolean right) {
        if (this.right) {
            return new Point(true, false);
        }

        return new Point(false, right);
    }
```